### DISCUSSION TITLE
AttributeError: 'NoneType' object has no attribute 'items'

### DISCUSSION THREAD

**Discussion by berto0** [StuffAnThings/qbit_manage#718] [Q&A] [answered]
Hi,
I've started using qbit manager, but I can't get it to work, or even connect to qbittorrent. (I have the qb URL, user and pass correctly).

The following appears in the log:

|     Version: 4.1.14 (Docker)                                                                       |
|     Platform: Linux-6.6.45-aarch64-with                                                            |
|     Scheduled Mode: Running every 1 day.                                                           |
|====================================================================================================|
|                                            Starting Run                                            |
|====================================================================================================|
| Locating config...                                                                                 |
| Using /config/config.yml as config                                                                 |
| Commands found in config.yml, ignoring env variables and using config commands instead.            |
|====================================================================================================|
|                                        DOCKER ENV COMMANDS                                         |
|====================================================================================================|
|     --run (QBT_RUN): False                                                                         |
|     --schedule (QBT_SCHEDULE): 1440                                                                |
|     --startup-delay (QBT_STARTUP_DELAY): 0                                                         |
|     --config-file (QBT_CONFIG): ['config.yml']                                                     |
|     --log-file (QBT_LOGFILE): activity.log                                                         |
|     --log-level (QBT_LOG_LEVEL): DEBUG                                                             |
|     --log-size (QBT_LOG_SIZE): 10                                                                  |
|     --log-count (QBT_LOG_COUNT): 5                                                                 |
|     --divider (QBT_DIVIDER): =                                                                     |
|     --width (QBT_WIDTH): 100                                                                       |
|     --debug (QBT_DEBUG): False                                                                     |
|     --trace (QBT_TRACE): False                                                                     |
|================================== CONFIG OVERRIDE RUN COMMDANDS ===================================|
|     --cross-seed (QBT_CROSS_SEED): False                                                           |
|     --recheck (QBT_RECHECK): False                                                                 |
|     --cat-update (QBT_CAT_UPDATE): False                                                           |
|     --tag-update (QBT_TAG_UPDATE): False                                                           |
|     --rem-unregistered (QBT_REM_UNREGISTERED): True                                                |
|     --tag-tracker-error (QBT_TAG_TRACKER_ERROR): False                                             |
|     --rem-orphaned (QBT_REM_ORPHANED): False                                                       |
|     --tag-nohardlinks (QBT_TAG_NOHARDLINKS): False                                                 |
|     --share-limits (QBT_SHARE_LIMITS): False                                                       |
|     --skip-cleanup (QBT_SKIP_CLEANUP): False                                                       |
|     --skip-qb-version-check (QBT_SKIP_QB_VERSION_CHECK): False                                     |
|     --dry-run (QBT_DRY_RUN): True                                                                  |
|====================================================================================================|
 | Traceback (most recent call last):                                                                 |
 |   File "/app/qbit_manage.py", line 482, in start                                                   |
 |     cfg = Config(default_dir, args)                                                                |
 |           ^^^^^^^^^^^^^^^^^^^^^^^^^                                                                |
 |   File "/app/modules/config.py", line 148, in __init__                                             |
 |     for tracker_urls, data in trackers.items():                                                    |
 |                               ^^^^^^^^^^^^^^                                                       |
 | AttributeError: 'NoneType' object has no attribute 'items'                                         |
 |                                                                                                    |
 | 'NoneType' object has no attribute 'items'                                                         |
 | Exiting scheduled Run.                            
|     Scheduled Mode: Running every 1 day.                                                           |
|====================================================================================================|
|                                            Finished Run                                            |
|                                         Run Time: 0:00:00                                          |

The config file is the following (I have it translated into Spanish, but before translating it it also did the same thing to me):

# Este es un archivo de configuración de ejemplo que documenta todas las opciones.
# Será necesario modificarlo para su caso de uso específico.
# Consulte el enlace a continuación para obtener más detalles sobre cómo configurar el archivo de configuración
# https://github.com/StuffAnThings/qbit_manage/wiki/Config-Setup

commands:
  # Los comandos definidos a continuación IGNORARÁ cualquier comando utilizado en la línea de comandos y en las variables de entorno de Docker.
  dry_run: True
  cross_seed: False
  recheck: False
  cat_update: False
  tag_update: False
  rem_unregistered: True
  tag_tracker_error: False
  rem_orphaned: False
  tag_nohardlinks: False
  share_limits: False
  skip_qb_version_check: False
  skip_cleanup: False

qbt:
  # Parámetros de qBittorrent
  host: *******************
  user: *********
  pass: **********

settings:
  force_auto_tmm: False # Obligará a qBittorrent a habilitar la Gestión Automática de Torrents para cada torrent.
  force_auto_tmm_ignore_tags: #Los torrents con estas etiquetas se ignorarán cuando force_auto_tmm esté habilitado.
  #  - cross-seed
  #  - Upload
  tracker_error_tag: issue # Establecerá la etiqueta de cualquier torrent que no tenga un rastreador en funcionamiento.
  nohardlinks_tag: noHL # Establecerá la etiqueta de cualquier torrente que no tenga enlaces físicos.
  share_limits_tag: ~share_limit # Agregaré esta etiqueta al aplicar límites de uso compartido para proporcionar una manera fácil de filtrar torrents por grupo de límite de uso compartido/prioridad para cada torrent.
  share_limits_min_seeding_time_tag: MinSeedTimeNotReached # Etiqueta que se agregará a los torrents que aún no hayan alcanzado el tiempo mínimo de propagación
  share_limits_min_num_seeds_tag: MinSeedsNotMet # Etiqueta que se agregará a los torrents que aún no hayan alcanzado el número mínimo de semillas
  share_limits_last_active_tag: LastActiveLimitNotReached # Etiqueta que se agregará a los torrents que aún no hayan alcanzado el último límite activo
  cross_seed_tag: cross-seed # Establecerá la etiqueta de cualquier torrente que se agregue mediante el comando cross-seed
  cat_filter_completed: True # Filtros para torrents completados solo cuando se ejecuta el comando cat_update
  share_limits_filter_completed: True # Filtros para torrents completados solo cuando se ejecuta el comando share_limits
  tag_nohardlinks_filter_completed: True # Filtros para torrents completados solo cuando se ejecuta el comando tag_nohardlinks
  cat_update_all: True # Comprueba y actualiza todas las categorías de torrents si se establece en Verdadero al ejecutar el comando cat_update; de ​​lo contrario, solo actualiza los torrents que no están categorizados
  disable_qbt_default_share_limits: False # Permite que QBM gestione los límites de uso compartido al deshabilitar los límites de uso compartido predeterminados de qBittorrents. Solo está activo cuando el comando share_limits está configurado como Verdadero

directory:
  # No quite estos
  # Cross-seed var: </your/path/here/>  #Directorio de salida de semillas cruzadas
  # root_dir var: </your/path/here/>  # Directorio de descargas raíz utilizado para verificar archivos huérfanos, noHL y RecycleBin.
  # <OPTIONAL> remote_dir var: </your/path/here/>  # Ruta de mapeo del host de Docker de root_dir.
  # remote_dir debe configurarse si está ejecutando qbit_manage localmente y qBittorrent/cross_seed está en un contenedor
  # remote_dir no debe configurarse si qbit_manage se ejecuta en un contenedor
  # <OPTIONAL> recycle_bin var: </your/path/here/>   # Ruta de la carpeta Papelera de reciclaje. La ubicación predeterminada es remote_dir/.RecycleBin
  # <OPTIONAL> torrents_dir var: </your/path/here/>  # Ruta del directorio de torrents de qbittorrent. Necesario para el atributo `save_torrents` en la papelera de reciclaje
  # <OPTIONAL> orphaned_dir var: </your/path/here/>  # Ruta de la carpeta de datos huérfanos. Es similar a la papelera de reciclaje, pero solo para datos huérfanos.
  cross_seed: 
  root_dir: /downloads/HDD/Descargas/Sonarr
  remote_dir: ""
  recycle_bin: /downloads/HDD/Papelera/RecycleBin
  torrents_dir: /qbittorrent/qBittorrent\BT_backup
  orphaned_dir: "" #/downloads/HDD/Papelera/orphaned_data

cat:
  # Parámetros de categoría y ruta
  # Todas las rutas de guardado en qbittorent deben completarse a continuación.
  # Si desea dejar una ruta de guardado como no categorizada, puede usar la clave 'Uncategorized' como nombre de la categoría.
  # <Category Name> : <save_path>  # Ruta de su directorio de guardado.
  Radarr: "/downloads/HDD/Descargas/Radarr"
  Sonarr: "/downloads/HDD/Descargas/Sonarr"
  HDOlimpo: "/downloads/HDD1/HDOlimpo"
  Torrentland: "/downloads/HDD1/Torrentland"
  HDZero: "/downloads/HDD2/HDZero"
  Xbytesv2: "/downloads/HDD3/Xbytesv2"

cat_change:
  # Esto mueve todos los torrents de una categoría a otra. Esto se ejecuta en --cat-update
  # ADVERTENCIA: si las rutas son diferentes y el modo de administración de torrents predeterminado está configurado en automático, los archivos podrían moverse.
  # <Old Category Name> : <New Category>
  ##Radarr-HD.cross-seed: movies-hd
  ##Radarr-UHD.cross-seed: movies-uhd
  ##movies-hd.cross-seed: movies-hd
  ##movies-uhd.cross-seed: movies-uhd

tracker:
  # Obligatorio
  # Parámetros de la etiqueta
  # <Tracker URL Keyword>:    #<OBLIGATORIO> Esta es la palabra clave en la URL del rastreador. Puede definir múltiples URL de rastreador dividiéndolas con el delimitador `|`
  # <OBLIGATORIO> Establezca el nombre de la etiqueta. Puede ser una lista de etiquetas o una sola etiqueta
  #   tag: <Tag Name>
  # <OPTIONAL> Establezca la categoría en función de la URL del rastreador. Esta opción de categoría tiene prioridad sobre la categoría definida por el directorio de guardado
  #   cat: <Category Name>
  # <OPTIONAL> Establezca esto en el nombre de la reacción de Notifiarr. Esto se utiliza para agregar reacciones del indexador a las notificaciones enviadas por Notifiarr.
  #   notifiarr: <notifiarr indexer>

nohardlinks:
  # Etiquetar películas o series que no tengan enlaces fijos fuera del directorio raíz
  # Es obligatorio completar el parámetro de directorio anterior para usar esta función (root_dir/remote_dir)
  # Esta variable debe configurarse con el nombre de la categoría de las películas o series completadas en qbit. La variable aceptable puede ser cualquier categoría que desee etiquetar si no se encuentran enlaces fijos
#  Radarr:
     # <OPTIONAL> exclude_tags var: Excluirá los torrents con cualquiera de las siguientes etiquetas al buscar en la categoría.
#    exclude_tags:
  ##    - Beyond-HD
  ##    - AnimeBytes
  ##    - MaM
    # <OPTIONAL> ignore_root_dir var: Ignorará cualquier enlace físico detectado en el mismo root_dir (valor predeterminado: Verdadero).
#    ignore_root_dir: true
  # Se pueden establecer categorías adicionales con tiempos de siembra/proporción definidos por separado.
#  series-completed:
    # <OPTIONAL> exclude_tags var: Excluirá los torrents con cualquiera de las siguientes etiquetas al buscar en la categoría.
#    exclude_tags:
      - Beyond-HD
      - BroadcasTheNet
    # <OPTIONAL> ignore_root_dir var: Ignorará cualquier enlace físico detectado en el mismo root_dir (valor predeterminado: Verdadero).
#    ignore_root_dir: false

share_limits:
  # Controla cómo se establecen los límites de uso compartido de torrents según la prioridad de tu agrupación
  # Cada torrent se emparejará con el grupo de límite de uso compartido con la prioridad más alta que cumpla con los criterios de filtro de grupo.
  # Cada torrent solo se puede emparejar con un grupo de límite de uso compartido
  # Esta variable es obligatoria y es un texto que define el nombre de tu agrupación. Puede ser cualquier cadena que desees
  noHL:
    # <MANDATORY> priority: <int/float> # Esta es la prioridad de su agrupación. Cuanto menor sea el número, mayor será la prioridad.
    priority: 1
    # <OPTIONAL> include_all_tags: <list> # Filtrar el grupo en función de una o más etiquetas. Se comprueban varias etiquetas include_all_tags con una condición AND
#   # Todas las etiquetas definidas aquí deben estar presentes en el torrent para que se incluya en este grupo
    include_all_tags:
    ##  - noHL
    # <OPTIONAL> include_any_tags: <list> # Filtrar el grupo en función de una o más etiquetas. Se seleccionan varias etiquetas include_any_tags con una condición OR
    # Cualquier etiqueta definida aquí debe estar presente en el torrent para que se incluya en este grupo.
    include_any_tags:
      - noHL
    # <OPTIONAL> exclude_all_tags: <list> # Filtrar excluyendo una o más etiquetas. Múltiples exclusiones_all_tags se verifican con una condición AND
    # Esto es útil para combinar con el filtro de categorías para excluir una o más etiquetas de una categoría completa
    # Todas las etiquetas definidas aquí deben estar presentes en el torrent para que se excluya de este grupo
    exclude_all_tags:
    ##  - Beyond-HD
    # <OPTIONAL> exclude_any_tags: <list> # Filtrar excluyendo una o más etiquetas. Se seleccionan varias etiquetas de exclusión con una condición OR
    # Esto es útil para combinarlo con el filtro de categorías para excluir una o más etiquetas de una categoría completa
    # Cualquier etiqueta definida aquí debe estar presente en el torrent para que se excluya de este grupo
    exclude_any_tags:
    ##  - Beyond-HD
    # <OPTIONAL> categories: <list> # Filtrar incluyendo una o más categorías. Las categorías múltiples se verifican con una condición OR
    # Dado que un torrent solo se puede asociar con una sola categoría, se verifican varias categorías con una condición OR
    categories:
      - Radarr
    # <OPTIONAL> max_ratio <float>: Establecerá la proporción máxima de compartición del torrent hasta que se detenga la propagación o carga del torrent y se podrá limpiar o eliminar si se han alcanzado los mínimos.
    # El valor predeterminado será -1 (sin límite) si no se especifica para el grupo.
##    max_ratio: 5.0
    # <OPTIONAL> max_seeding_time <str>: Establecerá el tiempo máximo de propagación del torrent hasta que se detenga la propagación/carga del torrent y se podrá limpiar/eliminar si se han alcanzado los mínimos.
    # Ver algunos ejemplos de expresiones de tiempo válidas (https://github.com/onegreyonewhite/pytimeparse2)
    # 32m, 2h32m, 3d2h32m, 1w3d2h32m
    # El valor predeterminado será -1 (sin límite) si no se especifica nada para el grupo. (Valor máximo de 1 año (525600 minutos))
    max_seeding_time: 7501m
    # <OPTIONAL> min_seeding_time <str>: Evitará la eliminación de torrent mediante la variable de limpieza si el torrent aún no ha alcanzado el tiempo mínimo de propagación (minutos).
    # Esto solo debe configurarse si lo está utilizando junto con max_seeding_time y max_ratio. Si no está configurando un max_ratio, utilice max_seeding_time en su lugar.
    # Si el torrent aún no ha alcanzado este tiempo mínimo de propagación, cambiará los límites de uso compartido a cero y reanudará el torrent para continuar la propagación.
    # Vea algunos ejemplos de expresiones de tiempo válidas (https://github.com/onegreyonewhite/pytimeparse2)
    # 32m, 2h32m, 3d2h32m, 1w3d2h32m
    # El valor predeterminado será 0 si no se especifica para el grupo.
    min_seeding_time: 7500m
    # <OPTIONAL> last_active <str>: Evitará la eliminación de torrent mediante la variable de limpieza si el torrent ha estado activo en los últimos x minutos.
    # Si el torrent ha estado activo en los últimos x minutos, cambiará los límites de uso compartido a cero y reanudará el torrent para continuar difundiendo.
    # Ver algunos ejemplos de expresiones de tiempo válidas (https://github.com/onegreyonewhite/pytimeparse2)
    # 32m, 2h32m, 3d2h32m, 1w3d2h32m
    # El valor predeterminado será 0 si no se especifica para el grupo.
##    last_active: 30d
    # <OPTIONAL> Limit Upload Speed <int>: Limitará la velocidad de carga KiB/s (KiloBytes/segundo) (`-1` : Sin límite)
    limit_upload_speed: -1
    # <OPTIONAL> Habilitar velocidad de carga de grupo <bool>: los límites de velocidad de carga se aplican a nivel de grupo. Esto tomará el valor de limit_upload_speed definido y lo dividirá equitativamente entre la cantidad de torrents en el grupo.
    enable_group_upload_speed: false
    # <OPTIONAL> cleanup <bool>: WARNING!! Si se establece como verdadero, se eliminarán y borrarán los contenidos de cualquier torrent que cumpla con los límites de uso compartido (tiempo máximo O proporción máxima).
    cleanup: true
    # <OPTIONAL> resume_torrent_after_change <bool>: Esta variable reanudará la descarga de tu torrent después de cambiar los límites de uso compartido. El valor predeterminado es verdadero
    resume_torrent_after_change: true
    # <OPTIONAL> add_group_to_tag <bool>: Esto agrega su agrupación como una etiqueta con un prefijo definido en la configuración. El valor predeterminado es verdadero
    # Ejemplo: una agrupación definida como noHL tendrá una etiqueta establecida en ~share_limit.noHL (si se usa el prefijo predeterminado)
    add_group_to_tag: true
    # <OPTIONAL> min_num_seeds <int>: Evitará la eliminación del torrent mediante la variable de limpieza si la cantidad de semillas es menor que el valor establecido aquí.
    # Si el torrent tiene una cantidad de semillas menor que min_num_seeds, los límites de uso compartido se cambiarán nuevamente a cero y se reanudará el torrent para continuar difundiendo.
    # El valor predeterminado será 0 si no se especifica para el grupo.
    min_num_seeds: 0
    # <OPTIONAL> custom_tag <str>: Aplique un nombre de etiqueta personalizado para este grupo en particular. **ADVERTENCIA (Esta etiqueta DEBE ser única, ya que se utilizará para determinar los límites de uso compartido. Asegúrese de que no se superponga con ninguna otra etiqueta en qbt)**
    custom_tag: sharelimits_noHL
  cross-seed:
    priority: 2
    include_all_tags:
      - cross-seed
    max_seeding_time: 7d
    cleanup: false
  PTP:
    priority: 3
    include_all_tags:
      - PassThePopcorn
    max_ratio: 2.0
    max_seeding_time: 90d
    cleanup: false
  default:
    priority: 999
    max_ratio: -1
    max_seeding_time: -1
    cleanup: false

recyclebin:
  # El método de eliminación de la papelera de reciclaje moverá los archivos a la papelera de reciclaje (ubicada en /root_dir/.RecycleBin) en lugar de eliminarlos directamente en qbit
  # De manera predeterminada, la papelera de reciclaje se vaciará cada vez que se ejecute el script qbit_manage si se define empty_after_x_days.
  enabled: true
  # <OPTIONAL> empty_after_x_days var:
  # Eliminará automáticamente todos los archivos y carpetas de la papelera de reciclaje después de x días. (Comprueba cada ejecución del script)
  # Si esta variable no está definida, la papelera de reciclaje nunca se vaciará.
  # ADVERTENCIA: ¡Si se establece esta variable en 0, se eliminarán todos los archivos inmediatamente después de ejecutar el script!
  empty_after_x_days: 2
  # <OPTIONAL> save_torrents var:
  # Si esta opción está configurada como verdadera, DEBE completar el atributo torrents_dir en el directorio.
  # Esto guardará una copia de su archivo .torrent y .fastresume en la papelera de reciclaje antes de eliminarlo de qbittorrent
  save_torrents: true
  # <OPTIONAL> split_by_category var:
  # Esto dividirá la carpeta de la papelera de reciclaje según la ruta de guardado definida en el atributo `cat`
  # y agregará el nombre de la carpeta base de la papelera de reciclaje que se definió en el subatributo `recycle_bin` bajo el directorio.
  split_by_category: false

orphaned:
  # Los archivos huérfanos son aquellos que se encuentran en el directorio de descarga root_dir y que no son referenciados por ningún torrent activo.
  # Eliminará automáticamente todos los archivos y carpetas en los datos huérfanos después de x días. (Verifica cada ejecución del script)
  # Si esta variable no está definida, los datos huérfanos nunca se vaciarán.
  # ADVERTENCIA: ¡Si se establece esta variable en 0, se eliminarán todos los archivos inmediatamente después de ejecutar el script!
  empty_after_x_days: 3
  # Patrones de archivos que no se considerarán archivos huérfanos. Útil para archivos generados que no forman parte del torrent pero pertenecen a los archivos del torrent
  exclude_patterns:
    - "**/.DS_Store"
    - "**/Thumbs.db"
    - "**/@eaDir"
    - "/data/torrents/temp/**"
    - "**/*.!qB"
    - "**/*_unpackerred"
  # Establezca el umbral deseado para la cantidad máxima de archivos huérfanos que qbm eliminará en una sola ejecución. (-1 para deshabilitar las protecciones)
  # Esto ayudará a reducir la cantidad de eliminaciones accidentales de grandes cantidades de archivos huérfanos en una sola ejecución
  # ADVERTENCIA: Establecer esta variable en -1 no brindará protección contra ninguna eliminación
  max_orphaned_files_to_delete: 50

apprise:
  # Integración de Apprise con webhooks
  # Déjelo en blanco para deshabilitarlo
  # Es obligatorio completar la URL del punto final de la API de Apprise
#  api_url: http://apprise-api:8000
  # Mandatory to fill out the notification url/urls based on the notification services provided by apprise. https://github.com/caronc/apprise/wiki
  notify_url:

notifiarr:
  # Notifiarr integration with webhooks
  # Leave Empty/Blank to disable
  # Mandatory to fill out API Key
  apikey: ####################################
  # <OPTIONAL> Set to a unique value (could be your username on notifiarr for example)
  instance:

webhooks:
  # Webhook notifications:
  # Possible values:
  # Set value to notifiarr if using notifiarr integration
  # Set value to apprise if using apprise integration
  # Set value to a valid webhook URL
  # Set value to nothing (leave Empty/Blank) to disable
#  error: https://mywebhookurl.com/qbt_manage
#  run_start: notifiarr
#  run_end: apprise
#  function:
#    cross_seed: https://mywebhookurl.com/qbt_manage
#    recheck: notifiarr
#    cat_update: apprise
#    tag_update: notifiarr
#    rem_unregistered: notifiarr
#    tag_tracker_error: notifiarr
#    rem_orphaned: notifiarr
#    tag_nohardlinks: notifiarr
#    share_limits: notifiarr
#    cleanup_dirs: notifiarr

thank you

**Comment by bakerboy448** [comment, marked as answer]
Your configure file is invalid and is missing required attributes.

Review the sample configure file for the correct structure and wiki for what is required.

You cannot simply arbitrarily delete config file sections.

You removed the mandatory and required tracker section.

**Reply by berto0** [reply]
Yes, I started over with the example config file, and I realized that one of the paths was not correct.
I configured and tested each change and finally got it working with a simple configuration.
Now I will gradually modify and test it.
Thank you very much.
